crates/wasmtime/src/config.rs

```diff
@@ -316,6 +316,14 @@ impl Config {
         self
     }
 
+    /// Configure whether async execution of WebAssembly will yield
+    /// whenever the "epoch" is incremented on the Engine via
+    /// [`Engine::increment_epoch`].
+    pub fn epoch_interruption(&mut self, enable: bool) -> &mut Self {
```

Comment:
Ideally we'll have one location for documenting epochs, what they're used for, and perhaps an example or two as well. I think that this `Config` method may be one of the better places to put the user-facing documentation. Ideally we'd have some sort of book chapter on all the methods of interrupting executing WebAssembly but for now I think `Config` methods will suffice. Could this get filled in with more high-level details and get an example?

(or really as long as the "canonical documentation" for epochs is in one place and everywhere else refers to that I think it's ok. I'm not sure this is something we've otherwise been diligent or great about with fuel or other various mechanisms implemented by Wasmtime)

Reply:
I wrote a long doc comment here and linked it from all the other method doc-comments (and expanded the others significantly too). Hopefully this is a bit more polished now.

No example yet but I can add that too if we think it's important -- I see that fuel has some toplevel examples in `examples/`; maybe that would be appropriate?

Reply:
The docs here look great to me, thanks!

But yeah if you'd be up for adding an epoch-equivalent example that'd be awesome.